rasa/engine/training/fingerprinting.py

```diff
@@ -33,11 +36,18 @@ def calculate_fingerprint_key(
     Returns:
         The fingerprint key.
     """
+    dependency_versions = {
+        package: pkg_resources.get_distribution(
+            import_name_to_package_map.get(package, package)
+        ).version
+        for package in graph_component_class.required_packages()
+    }
     fingerprint_data = {
         "node_name": rasa.utils.common.module_path_from_class(graph_component_class),
         "component_implementation": inspect.getsource(graph_component_class),
         "config": config,
         "inputs": inputs,
+        "dependency_versions": dependency_versions,
```

Comment:
this will invalidate any existing cache entries right? Not an issue but should prob be mentioned in the changelog entry.